File: Stack of Cards.java
class Solution {
    public String solution(String[] cards1, String[] cards2, String[] goal) {
        int index1 = 0, index2 = 0;
        for(String word : goal) {
            if(index1 < cards1.length && word.equals(cards1[index1])) {
                index1 += 1;
                continue;
            }
            
            if(index2 < cards2.length && word.equals(cards2[index2])) {
                index2 += 1;
                continue;
            }
            return "No";
        }
        return "Yes";
    }
}

File: Stack of Cards.c
#include <stdio.h>
#include <stdbool.h>
#include <string.h>

char* solution(char** cards1, int cards1_len, char** cards2, int cards2_len, char** goal, int goal_len) {
    int idx1 = 0, idx2 = 0, idx_goal = 0;

    while (idx_goal < goal_len) {
        if (idx1 < cards1_len && strcmp(cards1[idx1], goal[idx_goal]) == 0) {
            idx1++;
        } else if (idx2 < cards2_len && strcmp(cards2[idx2], goal[idx_goal]) == 0) {
            idx2++;
        } else {
            return "No";
        }
        idx_goal++;
    }

    return "Yes";
}